### DISCUSSION TITLE
Hard link doesn't work when osu! stable song folder is on a different drive

### DISCUSSION THREAD

**Discussion by hrfarmer** [ppy/osu#21902] [Q&A] [answered]
So I have my stable installation on my C drive, but in the .ini file I changed my songs folder to be on my D drive. Whenever I tried to import my maps in the setup wizard (with lazer being on my C drive as well), it still started duplicating all my beatmaps, which I cannot do because I will run out of storage. For now I'm just moving stable over to my D drive so I can do the hard link there, but I'm wondering if there is a fix for this.

**Comment by matte-ek** [comment, marked as answer]
Hard links across different drives is not possible due to a limitation of NTFS (the filesystem), this is nothing you can fix on a game level unfortunately.

**Reply by hrfarmer** [reply]
oh ok, there should probably be a warning/popup added to the setup then if it sees that they are on different drives

**Reply by bdach** [reply]
There already literally is one and it says

> A second copy of all files will be made during import. To avoid this, please make sure the lazer data folder is on the same drive as your previous osu! install (and the file system is NTFS).

**Reply by hrfarmer** [reply]
Thats not what it said for me though. When I had my osu stable installation (not songs folder), and my laser installation on my c drive, it said it would create a hard link and no extra disk space would be used. But when I ran that, it still started duplicating beatmaps, probably due to the reason matte brought up.

**Reply by bdach** [reply]
https://github.com/ppy/osu/issues/21904
